### PR TITLE
Use a Lua script for manual scan

### PR DESCRIPTION
**Pull request by yohanboniface**
First step in Lua world, yay.

I've another lua script idea, but this will be another PR (but this is why I've made the system a bit generic).

### CODE REVIEW

File: addok/helpers/collectors.py [hunk 1]
```diff
@@ -12,34 +13,28 @@ def only_commons(helper):
...
-                    tmp_keys = [first.db_key] + helper.filters
-                    DB.zinterstore(helper.pid, set(tmp_keys))
-                    ids = DB.zrevrange(helper.pid, 0, 500)
-                    DB.delete(helper.pid)
-                else:
-                    ids = DB.zrevrange(first.db_key, 0, 500)
-                others = [t.db_key for t in helper.tokens[1:]]
+                    # Always consider filters when doing manual intersect.
+                    keys = keys + helper.filters
+                    # But, hey, can we brut force again?
```
Comment (davidbgk) [inline addok/helpers/collectors.py]: `brute` :p

File: addok/helpers/scripts.py [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+from addok.db import DB
+
+
+@config.on_load
+def load_scripts():
+    root = Path(__file__).parent / 'lua'
+    for path in root.glob('*.lua'):
+        with path.open() as f:
+            name = path.name[:-4]
+            globals()[name] = DB.register_script(f.read())
```
Comment (davidbgk) [inline addok/helpers/scripts.py]: Not sure I'd put that under `globals`, imagine somebody creating a `return.lua` script or something like that 💣

**Comment by yohanboniface** [comment]
> Isn't it more pertinent to handle these special but common cases differently. I mean, returning suggestions on the rue de input is useless so do not return anything like that and wait for more input?

First point is that even if words are only common, we may have a way to reduce the candidate: either a filter (a citycode for example) or a latitude and longitude center, and in this case the results are pertinent.

Other point is that not returning anything until we have a "non common word", would mean no feedback for the 20 first characters when typing something like "rue de la sainte église". 
Even with "rue de", I think returning something is part of the process, because the feedback will be used right away by the user to feel comfortable that the computer is listening to them, and they will continue typing to better define the search.

File: addok/helpers/scripts.py (hunk 2 above)
Comment (yohanboniface) [inline addok/helpers/scripts.py]: Ahah, I knew that dealing with `globals` will create me troubles :p
Those scripts are commited into the core, there is not risk that a user create such a script by mistake and commit and does not see that the tests are broken and release a new version and… :)
I wanted a simple API like "scripts.xxxx", and we need to wait for the config to be loaded for the DB to be instanciated, to be able to register the scripts.

**Comment by davidbgk** [comment]
> Other point is that not returning anything until we have a "non common word", would mean no feedback

Well, maybe it's not about returning but fetching. I mean, the important point here is the visual feedback to the user that something is happening and if you want absolutely to display existing results you might want cache these very common case to avoid hitting your DB for that?

One other option might be to inform the user that terms are too common to return pertinent results and encourage him to type more. Or even pre-type/fill the input field:

1. typing `Ru`
2. the first input suggestion is `Rue de `
3. select and add `la s`
4. the first input selection is `Rue de la sainte`
5. and so on

I think that Google was using that approach some time ago(?), not sure it's still related to that issue though 😉

**Comment by yohanboniface** [comment]
> you might want cache these very common case to avoid hitting your DB for that?

Some people use Redis as cache, I use Redis as a DB. Should I have a cache anyway? ;)

> typing Ru
the first input suggestion is Rue de 

Keep in mind that there is a city called "Rue" in France. ;)
Also, there are plenty of "Rue Descartes" (or even "Rue Victor Hugo", i.e. without "de…"), so we cannot have this kind of short cuts imho.
Dealing with addresses, is like dealing with millions of edge cases, and we need to keep thing generic as much as possible because we can't anticipate all of those.

>  typing Ru
    the first input suggestion is Rue de
    select and add la s
    the first input selection is Rue de la sainte
    and so on

This suggests reviewing the behaviour of the API (clients should also adapt, etc.), I'm not against such brainstorming, but I think it's out of scope of this PR, which just takes the current behaviour and makes it faster :)

**Pull request merged**